### PR TITLE
Add domain to HTTP challenge errors

### LINKED ISSUE #8738: Include domain name for which "Cannot retrieve the ACME challenge for token XYZ: cannot find challenge for token XYZ"
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you expect to see?

I expect to see which domain is it related to (might be some typo in K8S environment deployment configuration or missing override of a template, which current useless log message doesn't help to find).

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

Add domain to the errors of the HTTP challenge.

### Motivation

Fixes #8738

### More

- [ ] ~~Added/updated tests~~
- [ ] ~~Added/updated documentation~~

### CODE REVIEW

File: pkg/provider/acme/challenge_http.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func (c *ChallengeHTTP) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 func (c *ChallengeHTTP) getTokenValue(ctx context.Context, token, domain string) []byte {
 	logger := log.FromContext(ctx)
-	logger.Debugf("Retrieving the ACME challenge for token %s...", token)
+	logger.Debugf("Retrieving the ACME challenge for %s token %s...", domain, token)
```
Comment (rtribotte) [inline pkg/provider/acme/challenge_http.go:106]:
```suggestion
	logger.Debugf("Retrieving the ACME challenge for domain %q and token %q...", domain, token)
```

File: pkg/provider/acme/challenge_http.go [hunk 2]
```diff
@@ -112,7 +112,7 @@ func (c *ChallengeHTTP) getTokenValue(ctx context.Context, token, domain string)
 		defer c.lock.RUnlock()
 
 		if _, ok := c.httpChallenges[token]; !ok {
-			return fmt.Errorf("cannot find challenge for token %s", token)
+			return fmt.Errorf("cannot find challenge for %s token %s", domain, token)
```
Comment (rtribotte) [inline pkg/provider/acme/challenge_http.go:115]:
```suggestion
			return fmt.Errorf("cannot find challenge for domain %q and token %q", domain, token)
```

File: pkg/provider/acme/challenge_http.go [hunk 3]
```diff
@@ -132,7 +132,7 @@ func (c *ChallengeHTTP) getTokenValue(ctx context.Context, token, domain string)
 	ebo.MaxElapsedTime = 60 * time.Second
 	err := backoff.RetryNotify(safe.OperationWithRecover(operation), ebo, notify)
 	if err != nil {
-		logger.Errorf("Cannot retrieve the ACME challenge for token %v: %v", token, err)
+		logger.Errorf("Cannot retrieve the ACME challenge for %s token %v: %v", domain, token, err)
```
Comment (rtribotte) [inline pkg/provider/acme/challenge_http.go:135]:
```suggestion
		logger.Errorf("Cannot retrieve the ACME challenge for domain %q and token %q: %v", domain, token, err)
```

**Review by rtribotte — CHANGES_REQUESTED** [review]
Quoting would be a good thing for readability, WDYT?

**Review by kevinpollet — APPROVED** [review]
LGTM

**Review by rtribotte — APPROVED** [review]
LGTM 👌

**Review by mpl — APPROVED** [review]

**Pull request merged**